### ISSUE TITLE
Seeding

### ISSUE THREAD

**Issue by basxto** [Praytic/youtd2#414]
For doing a high-score match on two different devices with the same waves, it would be practical to a) have the random waves, item drops and random gamemodes be derived from a seed in a reproducible manner b) see the current seed in "Game Stats" c) start single-player with a custom seed.

That might even be interesting for testing, since the same inputs would always lead to the same results.

**Comment by Kvel2D** [comment]
Having same randomness for wave stats and for tower distribution after waves is doable.

I think there's no point in using a seed for item drops because there are abilities which affect item chance and item quality.

**Comment by basxto** [comment]
That sounds basically okay. Shattered Pixel Dungeon also diverges with different input, but everybody had the same opportunities basically. Different builders influence item quality anyways.

> earlier/later before wave finish

Does that mean it depends on the elapsed game time? Or only during which wave it gets done?

If the input is humanly reproducible it would also allow speedrunning seeds, though not if it depends on elapsed time.

**Comment by Kvel2D** [comment]
No, tower distribution doesn't depend on game time. It depends on element levels and wave level.
So if player 1 researches ice level 3 before wave 10 finish and player 2 researches ice level 3 after wave 10 finish there are two different conditions and two different random results for towers distributed after wave 10:

1. wave 10 + ice level 3
2. wave 10 + ice level 2
